src/main/java/com/noroff/mefit/data/model/GoalWorkout.java

```diff
@@ -0,0 +1,37 @@
+package com.noroff.mefit.data.model;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.format.annotation.DateTimeFormat;
+
+import javax.persistence.*;
+import java.sql.Timestamp;
+
+@Entity
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+@Setter
+@Table(name = "goal_workout", schema = "public")
+public class GoalWorkout {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    public Long id;
+
+    @Column
+    @DateTimeFormat(pattern = "dd/MM/yyyy")
+    private Timestamp end_date;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "workout_id")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private Workout workoutId;
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "goal_id")
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    private Goal goalId;
```

Comment:
This should be named to the object and not the ID, even if it is the ID that is passed by Hibernate. (Source: Craig & Greg)
```suggestion
    private Goal goal;
```

Reply:
Nice reminding me of that!